secrets/external.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/sapcc/go-bits/must"
+)
+
+func CheckPwCmdEnv() {
+	pwCmd := os.Getenv("OS_PW_CMD")
+	if pwCmd != "" && os.Getenv("OS_PASSWORD") != "" {
+		return
+	}
+	// Retrieve user's password from external command.
+	out, err := exec.Command("sh", "-c", pwCmd).Output()
```

Comment:
This should set `cmd.Stderr = os.Stderr` on the command object, to ensure that errors from the executed commands are visible to the user.